index.js

```diff
@@ -32,16 +32,33 @@ module.exports = (ms, options = {}) => {
 	add(parsed.hours, 'hour', 'h');
 	add(parsed.minutes, 'minute', 'm');
 
-	if (options.compact) {
+	if (options.separateMs || ms < 1000) {
 		add(parsed.seconds, 'second', 's');
-		return '~' + ret[0];
+		if (options.formatSubMs) {
```

Comment:
It works well, but it feels like the codebase starts to get a bit messy here? #28  would inherently fix this though, so I'm not sure that it's too big of a deal to address that immediately. I'm open to discussing that though, since resolving it now would deal with the tech debt.